packages/x-data-grid/src/hooks/features/listView/useGridListView.tsx

```diff
@@ -0,0 +1,119 @@
+} from './gridListViewSelectors';
+import { gridColumnDefinitionsSelector, gridVisibleColumnDefinitionsSelector } from '../columns';
+import { gridDimensionsSelector } from '../dimensions';
+import { useGridApiEventHandler } from '../../utils/useGridApiEventHandler';
+
+export type GridListViewState = { listColumn: GridStateColDef | undefined };
+
+const getListColumnWidth = (apiRef: React.MutableRefObject<GridPrivateApiCommunity>) => {
+  const dimensions = gridDimensionsSelector(apiRef.current.state);
+  const columns = gridVisibleColumnDefinitionsSelector(apiRef);
+  const actionsColumn = columns.find((col) => col.type === 'actions');
+  const viewportWidth = dimensions.viewportInnerSize.width;
+  const listColumnWidth = actionsColumn
+    ? viewportWidth - actionsColumn.computedWidth
+    : viewportWidth;
+  return listColumnWidth;
+};
+
+export const listViewStateInitializer: GridStateInitializer<
+  Pick<DataGridProcessedProps, 'unstable_listColumn'>
+> = (state, props, apiRef) => ({
+  ...state,
+  listView: {
+    listColumn: { ...props.unstable_listColumn, computedWidth: getListColumnWidth(apiRef) },
+  },
+});
+
+export function useGridListView(
+  apiRef: React.MutableRefObject<GridPrivateApiCommunity>,
+  props: Pick<DataGridProcessedProps, 'unstable_listColumn'>,
+) {
+  /**
+   * API METHODS
+   */
+  const getListColumn = React.useCallback<GridListViewApi['getListColumn']>(
+    (field) => {
+      const listColumn = gridListColumnSelector(apiRef.current.state);
+      if (listColumn?.field === field) {
+        return listColumn;
+      }
+
+      const columns = gridColumnDefinitionsSelector(apiRef);
+      return columns.find((col) => col.field === field && col.type === 'actions');
+    },
+    [apiRef],
+  );
+
+  const getListColumnIndex = React.useCallback<GridListViewApi['getListColumnIndex']>(
+    (field) => {
+      const columns = gridVisibleListColumnDefinitionsSelector(apiRef);
+      return columns.findIndex((col) => col.field === field);
+    },
+    [apiRef],
+  );
+
+  const listColumnApi: GridListViewApi = {
+    getListColumn,
+    getListColumnIndex,
+  };
+
+  useGridApiMethod(apiRef, listColumnApi, 'private');
+
+  /*
+   * EVENTS
+   */
+  const prevInnerWidth = React.useRef<number | null>(null);
+  const handleGridSizeChange: GridEventListener<'viewportInnerSizeChange'> = (
+    viewportInnerSize,
+  ) => {
+    if (prevInnerWidth.current !== viewportInnerSize.width) {
+      prevInnerWidth.current = viewportInnerSize.width;
+      apiRef.current.setState((state) => {
+        return {
+          ...state,
+          listView: {
+            ...state.listView,
+            listColumn: {
+              ...state.listView.listColumn,
+              computedWidth: getListColumnWidth(apiRef),
+            },
+          } as GridListViewState,
+        };
+      });
+    }
+  };
+
+  useGridApiEventHandler(apiRef, 'viewportInnerSizeChange', handleGridSizeChange);
+
+  /**
+   * EFFECTS
+   */
```

Comment:
Lint: don't use `/**` except for jsdoc comments. Also applies above at `API METHODS`.

Reply:
I do agree, but it's following the convention used in other feature hooks. We could do a sweep to standardise this afterwards?

Reply:
We update these when we modify code around them, but new code should be without the `/**`.

Reply:
Fair enough, updated https://github.com/mui/mui-x/pull/14393/commits/69867e88d0fef3ede59f19ad126e37e6899a2ca4